### ISSUE TITLE
Cannot create admin account (/ redirects to /login)

### ISSUE THREAD

**Issue by gdeans** [Wingysam/Christmas-Community#134]
I setup this docker on an RPi 4 with the default env file and it started fine.  Browsed to the `IP:port` and it redirects to `IP:port/login`.  There is no way to create an admin account per your instructions.  Tried using guest and the password I set in the env file but that doesn't work either.  Any ideas?

**Comment by Wingysam** [comment]
Try clearing your cookies for the site? That issue sounds familiar but I haven't experienced it in a year or two.

**Comment by gdeans** [comment]
Tried on 4 different devices with at least 2 browsers on each device.  So I think cookies or any client side issues have been eliminated.  I also tried opening a console on the container, but your docker image doesn't have bash so it's slow going.  I'm not familiar with what you used to put this together so any help would be appreciated.

**Comment by gdeans** [comment]
I've now also tried portable browsers (K-Melon & Libre Wolf) in a different local account profile with no settings or plugins changed and get the same results.  Attached is a copy of my compose and env files, but I don't think anything is wrong there.  I just need to stop the redirect.
[Test_ChristmasCommunity.zip](https://github.com/user-attachments/files/16234224/Test_ChristmasCommunity.zip)

**Comment by Guiraud** [comment]
Same issue here : 
cleaned the cookies..
cleaned the docker
changed navigators..
any clue ?

**Comment by Wingysam** [comment]
So it's reproducible by trying to set the instance up with no volumes mounted on the latest docker image?

**Comment by Guiraud** [comment]
yes, 
I suppressed most of the .env variables and it worked. 
keeping only language, markdown and Smile.
I'll narrow it down and come back to you :) must be a not RTFM behaviour on my part.

**Comment by Guiraud** [comment]
Ok, obviously it's all about `DEFAULT_FAILURE_REDIRECT=/login`.
At the first login for the admin creation, this value should not be set.

But it would make sense to set this value after admin password creation.

I would recommend to set it afterward with the help of this [stackoverflow redis answer](https://stackoverflow.com/questions/10676951/listen-for-changes-in-redis/10677147#10677147)

Although I think it is updated after first login.
It should just be commented out in the [Readme.md](https://github.com/Wingysam/Christmas-Community/blob/master/README.md) file.

**Comment by gdeans** [comment]
I commented the env file out in the docker compose yml and it worked.  I added it back with only `LANGUAGE` defined in the env file and it worked again, but the CSS rendering was different for the setup page.  Also the image doesn't seem to honor the `PUID`/`PGID` being set in the compose file and runs everything as root.  I assume the container is running the web server as root or it's running it's setup routines as root and not changing the permissions to match the config.  I'm deleting the `secrets.txt` and `dbs` folder before each test as well to make sure nothing passes from the previous test.

**Comment by gdeans** [comment]
OK, something is just not right here.  I'm back to my original setup and it's working so there must be something outside the container or removing the container in portainer didn't wipe everything for some reason.

I did find out if I chown the data folder to the `PUID`/`PGID` it still works, so I think whatever process sets the structure up is a one-time thing run as root and it doesn't seem to matter afterwards, which is also odd.

**Comment by Wingysam** [comment]
Do you have a way to make it reliably fail? Something is definitely wrong and I'd like to fix whatever caused that or at least make the app tell you what's wrong if it's a permissions issue or something.

**Comment by servebeer** [comment]
If by any chance you are using Unraid, there’d is an issue with the App Store temples. Edit the template, enable advanced view, look for the WebUI: line and add /login to the end. Also, after creating the initial admin account, look for “Login Failure Redirect:“ and change that to /login as well.

**Comment by gdeans** [comment]
Nope, regular Docker on debian.  Still need to test a clean setup on my other box.  I'll try to get to it this week.
